desktop-widgets/mapwidget.cpp

```diff
@@ -94,7 +93,15 @@ void MapWidget::selectionChanged()
 void MapWidget::selectedDivesChanged(const QList<int> &list)
 {
 	CHECK_IS_READY_RETURN_VOID();
-	MainWindow::instance()->diveList->selectDives(list);
+	// We get a list of dive indices, but the selection code wants a list of dives.
+	// Therefore, transform them here.
+	std::vector<dive *> selection;
+	selection.reserve(list.size());
+	for (int idx: list) {
+		if (dive *d = get_dive(idx))
+			selection.push_back(d);
+	}
+	setSelection(selection, current_dive);
```

Comment:
this is the first time i see:
https://github.com/Subsurface-divelog/subsurface/blob/7387b10b2944c0126999a3911d6ecf705d243a8b/core/selection.cpp#L151

if this now the way to change currentDive, LGTM.

Reply:
so you're good with merging this, @neolit123 ?

Reply:
yes